pwndbg/lib/cache.py

```diff
@@ -138,7 +138,13 @@ def decorator(*a: P.args, **kw: P.kwargs) -> T:
                 # Check if the value is in the cache; if we have a cache miss,
                 # we return a special singleton object `_NOT_FOUND_IN_CACHE`. This way
                 # we can also cache a result of 'None' from a function
-                cached_value: Any = cache.get(key, _NOT_FOUND_IN_CACHE)
+                no_cache = False
+                try:
+                    cached_value: Any = cache.get(key, _NOT_FOUND_IN_CACHE)
+                except TypeError:
+                    # skip caching unhashable arguments
```

Comment:
When does this occur? Which functions? Can we make some exception just for them, like, another decorator that would try except? The cache function is executed many many times and I'd prefer to not add additional logic if not needed

Reply:
I need it because `read_reg` can take in an optional `gdb.Frame` which is unhashable. I could rewrite the code so that there is only extra logic in the `except` case, which should have almost no performance hit, [at least since Python 3.11](https://github.com/python/cpython/issues/84403).